sdk/go/common/resource/plugin/provider.go

```diff
@@ -217,6 +218,85 @@ type DiffResult struct {
 	DeleteBeforeReplace bool                    // if true, this resource must be deleted before recreating it.
 }
 
+// Computes the detailed diff of Updated, Added and Deleted keys.
+func NewDetailedDiffFromObjectDiff(diff *resource.ObjectDiff) map[string]PropertyDiff {
+	if diff == nil {
+		return map[string]PropertyDiff{}
+	}
+	return objectDiffToDetailedDiff("", diff)
+}
+
+func objectDiffToDetailedDiff(prefix string, diff *resource.ObjectDiff) map[string]PropertyDiff {
```

Comment:
Nit: I think passing in an accumulator for the result in these functions will simplify the code by removing the need for the various copy loops.

```suggestion
func objectDiffToDetailedDiff(result map[string]PropertyDiff, prefix string, diff *resource.ObjectDiff) {
```